File: src/org/pcsoft/plugins/intellij/iss/language/IssFileType.java
package org.pcsoft.plugins.intellij.iss.language;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.pcsoft.plugins.intellij.iss.IssIcons;

import javax.swing.*;

/**
 * Created by Christoph on 30.09.2016.
 */
public class IssFileType extends LanguageFileType {
    public static final IssFileType INSTANCE = new IssFileType();

    public IssFileType() {
        super(IssLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Inno Setup Script";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Inno Setup Script File";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "iss";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return IssIcons.FILE;
    }
}

File: src/main/kotlin/org/pcsoft/intellij/plugin/inno_setup/language/IssFileType.kt
/*
 * Copyright (c) KleinerHacker alias Pfeiffer C Soft 2026.
 * This work is licensed under the Apache License, Version 2.0.
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at:
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, this software is distributed on an “AS IS” BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and limitations.
 */

package org.pcsoft.intellij.plugin.inno_setup.language

import com.intellij.openapi.fileTypes.LanguageFileType
import org.pcsoft.intellij.plugin.inno_setup.IssIcons
import javax.swing.Icon

class IssFileType : LanguageFileType(IssLanguage) {
    companion object {
        @JvmField
        val INSTANCE = IssFileType()
    }

    override fun getName() = "Inno Setup Script"
    override fun getDescription() = "Inno Setup script file"
    override fun getDefaultExtension() = "iss"
    override fun getIcon(): Icon = IssIcons.ScriptFile
}
